proto/src/google/protobuf/duration.rs

```diff
@@ -0,0 +1,314 @@
+    /// of the same sign as the `seconds` field. Must be from -999,999,999
+    /// to +999,999,999 inclusive.
+    #[prost(int32, tag = "2")]
+    pub nanos: i32,
+}
+
+impl Name for Duration {
+    const PACKAGE: &'static str = PACKAGE;
+    const NAME: &'static str = "Duration";
+
+    fn type_url() -> String {
+        type_url_for::<Self>()
+    }
+}
+
+const NANOS_PER_SECOND: i32 = 1_000_000_000;
+const NANOS_MAX: i32 = NANOS_PER_SECOND - 1;
+
+impl Duration {
+    /// Normalizes the duration to a canonical format.
+    pub fn normalize(&mut self) {
+        // Make sure nanos is in the range.
+        if self.nanos <= -NANOS_PER_SECOND || self.nanos >= NANOS_PER_SECOND {
+            if let Some(seconds) = self
+                .seconds
+                .checked_add((self.nanos / NANOS_PER_SECOND) as i64)
+            {
+                self.seconds = seconds;
+                self.nanos %= NANOS_PER_SECOND;
+            } else if self.nanos < 0 {
+                // Negative overflow! Set to the least normal value.
+                self.seconds = i64::MIN;
+                self.nanos = -NANOS_MAX;
+            } else {
+                // Positive overflow! Set to the greatest normal value.
+                self.seconds = i64::MAX;
+                self.nanos = NANOS_MAX;
+            }
+        }
+
+        // nanos should have the same sign as seconds.
+        if self.seconds < 0 && self.nanos > 0 {
+            if let Some(seconds) = self.seconds.checked_add(1) {
+                self.seconds = seconds;
+                self.nanos -= NANOS_PER_SECOND;
+            } else {
+                // Positive overflow! Set to the greatest normal value.
+                debug_assert_eq!(self.seconds, i64::MAX);
+                self.nanos = NANOS_MAX;
+            }
+        } else if self.seconds > 0 && self.nanos < 0 {
+            if let Some(seconds) = self.seconds.checked_sub(1) {
+                self.seconds = seconds;
+                self.nanos += NANOS_PER_SECOND;
+            } else {
+                // Negative overflow! Set to the least normal value.
+                debug_assert_eq!(self.seconds, i64::MIN);
+                self.nanos = -NANOS_MAX;
+            }
+        }
+    }
+}
+
+/// Converts a `core::time::Duration` to a `Duration`.
+impl From<core::time::Duration> for Duration {
+    fn from(duration: core::time::Duration) -> Duration {
+        let seconds = duration.as_secs();
+        let seconds = if seconds > i64::MAX as u64 {
+            i64::MAX
+        } else {
+            seconds as i64
+        };
+        let nanos = duration.subsec_nanos();
+        let nanos = if nanos > i32::MAX as u32 {
+            i32::MAX
+        } else {
+            nanos as i32
+        };
+        let mut duration = Duration { seconds, nanos };
+        duration.normalize();
+        duration
+    }
+}
+
+impl TryFrom<Duration> for core::time::Duration {
+    type Error = core::time::Duration;
```

Comment:
Hmm, this previously returned `DurationError`: 

https://docs.rs/prost-types/0.13.1/src/prost_types/duration.rs.html#86-104

```rust
impl TryFrom<Duration> for time::Duration {
    type Error = DurationError;

    /// Converts a `Duration` to a `std::time::Duration`, failing if the duration is negative.
    fn try_from(mut duration: Duration) -> Result<time::Duration, DurationError> {
        duration.normalize();
        if duration.seconds >= 0 && duration.nanos >= 0 {
            Ok(time::Duration::new(
                duration.seconds as u64,
                duration.nanos as u32,
            ))
        } else {
            Err(DurationError::NegativeDuration(time::Duration::new(
                (-duration.seconds) as u64,
                (-duration.nanos) as u32,
            )))
        }
    }
}
```

Reply:
Good catch, will fix it! As well as replace the `From<core::time::Duration>` instance with a `TryFrom<core::time::Duration>` following `prost-types`'s implementation instead of `pbjson-types` which I lifted those from.

Reply:
https://github.com/informalsystems/tendermint-rs/pull/1456